### ISSUE TITLE
New Property "MaxFontSize" 

### ISSUE THREAD

**Issue by darrabam** [dotnet/maui#21370]
### Description

Current challenge with having FontAutoScalingEnabled=True is that when the user sets his OS scaling to large values, it often mess up the interface, especially if you are explicitly setting elements size/heights. 

On the other hand, ignoring the OS scaling by setting FontAutoScalingEnabled=False is also a bad idea and wont make for a good user experience. 

With Xamarin, I used to have FontSizeConverter which relies on Device.GetNamedSize to get the scaled named font size and then cap it at my predefined limits. 

MAUI is marking Device.GetNamedSize as obsolete, so there is no way to get the scaled named font sizes. Therefore, I propose to have a new property (MaxFontSize) on all elements with text (Label, Button ....etc.), such that the scaled font shall not exceed MaxFontSize.

### Public API Changes

<Label Text="MyText" FontAutoScalingEnabled="True" FontSize="14" MaxFontSize="18" />

### Intended Use-Case

To cap the scaling of the font when FontAutoScalingEnabled="True" to a maximum value. The is useful when the OS scaling is higher than what the app/layout was designed for.

**Comment by awasilik** [comment]
Question - on iOS there is a property maximumContentSizeCategory, but it seem to be completely ignored by MAUI

https://developer.apple.com/documentation/uikit/uiview/3750923-maximumcontentsizecategory


Is there any reason for this?

**Comment by rudyspano** [comment]
@darrabam seems proposed here as well: https://github.com/dotnet/maui/issues/14438

**Comment by darrabam** [comment]
@rudyspano correct! the proposal in #14438 seems exactly like what I need, somehow I missed it in my research,
